applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+	struct data_fifo *msg_rx;
+
+	/* Pointer to a modules data transmitter fifo */
+	struct data_fifo *msg_tx;
+
+	/* Pointer to the data buffer slab */
+	struct k_mem_slab *data_slab;
+
+	/* Size of a data block taken from the slab */
+	size_t data_size;
+};
+
+/**
+ * @brief Module's generic set-up structure.
+ */
+struct amod_parameters {
+	/* The modules private description */
+	struct amod_description *description;
+
+	/* The modules thread setting */
+	struct amod_thread_configuration thread;
+};
+
+/**
+ * @brief Private module handle.
+ */
+struct amod_handle {
+	/* Unique name of this module instance */
+	char name[CONFIG_AMOD_NAME_SIZE];
+
+	/* The modules description */
+	struct amod_description *description;
+
+	/* Current state of the module */
+	enum amod_state state;
+
+	/* Previous state of the module */
+	enum amod_state previous_state;
+
+	/* Thread ID */
+	k_tid_t thread_id;
+
+	/* Thread data */
+	struct k_thread thread_data;
+
+	/* Flag to indicate the module should send data block to its TX fifo */
+	bool data_tx;
+
+	/* List node (next pointer) */
+	sys_snode_t node;
+
+	/* A singley linked-list of the handles the module is connected to */
+	sys_slist_t hdl_dest_list;
+
+	/* Number of destination modules */
+	uint8_t dest_count;
+
+	/* Semaphore to count messages between modules */
+	struct k_sem sem;
+
+	/* Mutex to make the above destinations list thread safe */
+	struct k_mutex dest_mutex;
+
+	/* Modules thread configuration */
+	struct amod_thread_configuration thread;
+
+	/* Private context for the module */
+	struct amod_context *context;
+};
+
+/**
+ * @brief  Pointer to a response callback function for a response to a data send as
+ *         supplied by the module user.
+ *
+ * @param handle  The handle of the module that sent the data message
+ * @param block  The size of the supporting data or 0 if none
+ */
+typedef void (*amod_response_cb)(struct amod_handle *handle, struct aobj_block *block);
+
+/**
+ * @brief Private structure describing a data in message into the module thread.
+ */
+struct amod_message {
+	/* Audio data block to input */
+	struct aobj_block block;
+
+	/* Sending module's handle */
+	struct amod_handle *tx_handle;
+
+	/* Callback for when the data has been consumed */
+	amod_response_cb response_cb;
+};
+
+/**
+ * @brief A helper structure for describing the modules in a stream as a table.
+ */
+struct amod_table {
+	/* A unique name for the module */
+	char *name;
+
+	/* A unique handle to the module */
+	struct amod_handle *handle;
+
+	/* A unique handle to the module */
+	struct amod_configuration *initial_config;
+
+	/* the modules parameters */
+	struct amod_parameters params;
+
+	/* A unique module context */
+	struct amod_context *context;
+};
+
+/**
+ * @brief  Function for opening a module.
+ *
+ * @param parameters     Pointer to the module set-up parameters
+ * @param configuration  A pointer to the modules configuration (this must also
```

Comment:
```suggestion
 * @param configuration  Pointer to the modules configuration (this must also
```

Need to have consistency